src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,144 @@
+                printList(0, taskCount);
+            } else if (AddToList.keywordCheck(inputStringSplit[0])) {
+                if (inputStringSplit.length > 1) {
+                    AddToList.AddToList(inputStringSplit[0], inputStringSplit[1]);
+                } else {
+                    DukeException.taskDescriptionEmpty();
+                }
+            } else {
+                DukeException.illegalInput();
+            }
+        }
+    }
+
+    private static void readFile(List<Task> lists){
+        try {
+            File file = new File("src/main/java/Duke/Duke.txt");
+            if (file.createNewFile()) {
+                System.out.println("A new file has been created! ^_^");
+            } else {
+                System.out.println("Reading saved Task Lists ^_^");
+                Scanner readingFile = new Scanner(file);
+                while (readingFile.hasNextLine()) {
+                    String line = readingFile.nextLine();
+                    String[] parts = line.split("\\|", 4);
+                    String type = parts[0];
+                    String isDone = parts[1];
+                    String task = parts[2];
+                    if (type.equals("T")) {
+                        Task taskInFile = new ToDoTask(task);
+                        lists.add(taskInFile);
+                        if (isDone.equals("true")) {
+                            taskInFile.markAsDone();
+                        }
+                    } else if (type.equals("D")) {
+                        String time = parts[3];
+                        DeadlineTask taskInFile = new DeadlineTask(task, time);
+                        lists.add(taskInFile);
+                        if (isDone.equals("true")) {
+                            taskInFile.markAsDone();
+                        }
+                    } else {
+                        String time = parts[3];
+                        EventTask taskInFile = new EventTask(task, time);
+                        lists.add(taskInFile);
+                        if (isDone.equals("true")) {
+                            taskInFile.markAsDone();
+                        }
+                    }
+                }
+                taskCount = lists.size();
+                printList(0, taskCount);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void writeFile(List<Task> lists){
+        try {
+            FileWriter writer = new FileWriter("src/main/java/Duke/Duke.txt",false);
+            for (Task taskInList : lists) {
+                if (taskInList instanceof ToDoTask) {
+                    writer.write(taskInList.getTaskType() + "|" + taskInList.isDone() + "|" + taskInList.getTask().trim());
+                } else {
+                    writer.write(taskInList.getTaskType() + "|" + taskInList.isDone() + "|" + taskInList.getTask().trim() + "|" + taskInList.getTaskTime().trim());
```

Comment:
If your code lines are getting overly long, it's best to break it down into multiple lines (e.g. you could save your variables before writing them)